lib/Module.js

```diff
@@ -101,8 +101,13 @@ class Module extends DependenciesBlock {
 		/** @type {object} */
 		this.buildInfo = undefined;
 
-		/** @type {boolean} */
+		// TODO refactor this -> options object filled from Factory
 		this.useSourceMap = false;
+		this.lineToLine = false;
+
+		// TODO figure out if this should be defined here instead of `NormalModule`.
+		// Without this, type checking fails because the hooks use `Module`.
+		this.resource = undefined;
```

Comment:
I think we should decide what to do here.
Most hooks use `Module` but a lot of plugins assume `NormalModule`.

Reply:
These cases are actually bugs. Not every Module has `resource`. When used it should check for `instanceof NormalModule`.

Reply:
Same for `lineToLine`, but I guess `lineToLine` can be removed in webpack 5 anyway. Nobody uses this...